### ISSUE TITLE
Behavior changes unexpectedly in 0.1.17

### ISSUE THREAD

**Issue by kulp** [dtolnay/paste#38]
Is there a known change in behavior in 0.1.17 ?

I see that the changes https://github.com/dtolnay/paste/compare/0.1.16...0.1.17 are significant in terms of implementation, but it is not obvious that there should be different behavior. However, I found in [`lightning-sys`](https://github.com/Petelliott/lightning-sys) that the macro expansion was significantly different.

- [Travis build failing with `paste "0.1"`](https://travis-ci.com/github/kulp/lightning-sys/builds/172354956)
- [Travis build working with `paste "=0.1.16"`](https://travis-ci.com/github/kulp/lightning-sys/builds/172355201)

As of https://github.com/Petelliott/lightning-sys/commit/ba3bffb9025292cf7eb4e09f0143096157f8e7b8 :
```bash
$ cargo update -p paste --precise 0.1.16
$ cargo expand --lib | sed -n '/^pub mod jitstate/,/^}/p' > with-0.1.16.txt
$ cargo update -p paste --precise 0.1.17
$ cargo expand --lib | sed -n '/^pub mod jitstate/,/^}/p' > with-0.1.17.txt
$ wc -l with-0.1.1*.txt
    4950 with-0.1.16.txt
    4229 with-0.1.17.txt
```
[with-0.1.16.txt](https://github.com/dtolnay/paste/files/4808756/with-0.1.16.txt)
[with-0.1.17.txt](https://github.com/dtolnay/paste/files/4808757/with-0.1.17.txt)

I have not boiled down a small test case.

**Comment by dtolnay** [comment]
Sorry about the breakage, this is fixed in 0.1.18.

**Comment by kulp** [comment]
I confirm. Thanks.
